src/gogma/overlap.jl

```diff
@@ -99,17 +99,36 @@ end
 
 function force!(f::AbstractVector, x::AbstractIsotropicGaussian, y::AbstractIsotropicGaussian,
                 s=x.σ^2+y.σ^2, w=x.ϕ*y.ϕ; coef=1)
-        return force!(f, x.μ, y.μ, s, coef*w)
+    return force!(f, x.μ, y.μ, s, coef*w)
 end
 
-function force!(f::AbstractVector, x::AbstractIsotropicGaussian, y::AbstractIsotropicGMM; kwargs...)
-    for gy in y.gaussians
-        force!(f, x, gy; kwargs...)
+function force!(f::AbstractVector, x::AbstractIsotropicGaussian, y::AbstractIsotropicGMM, pσ=nothing, pϕ=nothing; kwargs...)
+    if isnothing(pσ) && isnothing(pϕ)
+        pσ = x.σ^2 .+ [gy.σ^2 for gy in y.gaussians]
+        pϕ = x.ϕ * [gy.ϕ for gy in y.gaussians]
```

Comment:
```suggestion
        pϕ = [ x.ϕ * gy.ϕ for gy in y.gaussians]
```
This avoids allocation of a temporary array, going straight for the array you'll use for the force calculation.